File: NDPServer/src/main/java/com/slender/controller/AdministrationController.java
package com.slender.controller;

import com.slender.result.Response;
import com.slender.service.interfase.UserService;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@Tag(name = "管理员接口")
@RequestMapping("/admins")
@RequiredArgsConstructor
public class AdministrationController {
    private final UserService userService;
    @DeleteMapping("/block/{uid}")
    public Response<Void> block(@AuthenticationPrincipal Long uid,
                                @PathVariable("uid") Long userUid){
        userService.block(userUid);
        return Response.success("封禁成功");
    }
}

File: NDPServer/src/main/java/com/slender/controller/AdministrationController.kt
package com.slender.controller

import com.slender.result.Response
import com.slender.service.interfase.OrderService
import com.slender.service.interfase.ProductService
import com.slender.service.interfase.UserService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.Parameter
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@Tag(name = "管理员接口")
@RequestMapping("/admins")
open class AdministrationController(
    private val userService: UserService,
    private val orderService: OrderService,
    private val productService: ProductService
) {
    @DeleteMapping("/block/{uid}")
    @Operation(summary = "封禁用户")
    open fun block(
        @Parameter(hidden = true) @AuthenticationPrincipal uid: Long,
        @PathVariable("uid") userUid: Long
    ): Response<Unit> {
        userService.block(userUid)
        return Response.success("封禁成功")
    }

    @GetMapping("/orders/export")
    @Operation(summary = "导出订单")
    open fun exportOrders(response: HttpServletResponse) {
        orderService.export(response)
    }

    @GetMapping("/products/export")
    @Operation(summary = "导出产品")
    open fun exportProducts(response: HttpServletResponse) {
        productService.export(response)
    }
}